### ISSUE TITLE
delete_user issue

### ISSUE THREAD

**Issue by guillelopez** [plaid/plaid-ruby#54]
Hi, 

I am having an issue with `delete_user` in tartan: 

```
plaid_user = Plaid::User.new
plaid_user.access_token = "[USER_ACCESS_TOKEN]"
plaid_user.delete_user()
```

I get:  `<Net::HTTPUnauthorized 401 Unauthorized readbody=true>`

If I try straight with curl:

```
curl -X DELETE https://tartan.plaid.com/info \
  -d client_id=[CLIENT_ID]\
  -d secret=[CLIENT_SECRET} \
  -d access_token=[USER_ACCESS_TOKEN]
```

Then I get:
```
{
  "code": 1110,
  "message": "product not enabled",
  "resolve": "This product is not enabled for this item. Use the upgrade route to add it.",
  "access_token": "[USER_ACCESS_TOKEN]"
}
```

I imagine this is all happening because the new `info` endpoint. If I do that same call to `https://tartan.plaid.com/connect` then it works out fine.

Currently I am using `plaid-ruby version 1.4.3 `. On the README it says it's the stable version (?).

I tried as well with `plaid-ruby version 1.4.0` (the one that loads through `gem 'plaid'`) but as well there it's making the `delete_user` call to `/info`.

Could you guys provide me any light into this issue? Thanks.

**Comment by guillelopez** [comment]
Hi. Any update about this issue?

**Comment by j4ustin** [comment]
@guillelopez Have you tried using this on a live account? I agree, its likely due to the testing suite and the product being in "beta" still. I don't work for Plaid, so I can't be 100% sure.

**Comment by guillelopez** [comment]
@j4ustin I haven't tried. But I don't see any reason why it will behave in a different way.

**Comment by j4ustin** [comment]
@guillelopez Usually for these features there is a bit of lag time between release and them being inserted into the public facing test accounts. I've noticed this on a few Plaid features in the past. 

I would post this question directly to their tech team as it is not specific to the ruby gem.

**Comment by guillelopez** [comment]
Hi. It feels a little bit awkward that it's been 15 days and no one from Plaid is providing me any response... And don't get me wrong, this is not a big deal, but I always felt you guys where on top of these details. @michaelckelly ? @whockey ?

**Comment by michaelckelly** [comment]
Sorry for the delay, @guillelopez. 

Are you running into issues with a live account or a Plaid sandbox account?

If it it's with a live account, you need to be sure that you've added the info product to that user either via the initial user add (`POST /info`) or the upgrade endpoint (`POST /upgrade`).

Again, our apologies for the slow reply - we do our best to keep a close eye on our client library repositories but issues do sometimes slip between the cracks. Feel free to drop a note to <support@plaid.com> if this happens again - that's the quickest way to get a hold of someone from our support team!

**Comment by guillelopez** [comment]
Thanks @michaelckelly. 

I am having this issue calling to `tartan.plaid.com` but with a real account (not `plaid_test`).

I am going to try adding to the user the info product through `POST /upgrade` and I imagine then it will work fine. I will let you know the result. 

Once this is working, it might be good to update the info here: https://plaid.com/docs/#upgrade and here: https://plaid.com/docs/#delete-user

**Comment by guillelopez** [comment]
@michaelckelly on the process of checking if the `delete_user` works, I think I spotted another problem: 

It's here: https://github.com/plaid/plaid-ruby/blob/master/lib/plaid/connection.rb#L49

Are you sure a `Net::HTTP::Delete` instance is permited to use a request body?

When executing: 

```
plaid_user = Plaid::User.new
plaid_user.access_token = access_token
plaid_user.upgrade('info')
plaid_user.delete_user()
```

The response I get is: 
```
code: 1100
message: client_id missing
resolve: Include your Client ID so we know who you are.
```

`Plaid.customer_id` and `Plaid.secret` are properly initialized.

**Comment by michaelckelly** [comment]
@guillelopez-

We'll take a look into the issues with deleting a user, thanks for the heads up!

**Comment by soroushjp** [comment]
Hi @guillelopez, Soroush from Plaid here. I was able to do some investigating this weekend and made sure the request body was being passed with the DELETE request. Confirmed with an HTTP proxy that the body was being passed, but `delete_user()` was clearly still not working. Finally figured out that it was a wrong `Content-Type` header -- we needed a `application/x-www-form-urlencoded` header and the library was sending `multipart/form-data`.

It should be working with the fix in the `sjp-delete-issue` branch, could you checkout and confirm that it fixes the issue for you?

**Comment by soroushjp** [comment]
@guillelopez I've gone ahead merged the PR and closed this issue. Please let me know if it's still not working for you and we can investigate further.

**Comment by guillelopez** [comment]
Ok. I will give it a try this evening or tomorrow morning. Sorry I couldn't do it earlier.

**Comment by soroushjp** [comment]
@guillelopez No worries at all, we just wanted to push out this fix, but keep us posted on whether it solves the problem for you.

**Comment by guillelopez** [comment]
@soroushjp and @michaelckelly, I just tested it and everything is working fine. Thanks for taking care of it.

**Comment by soroushjp** [comment]
@guillelopez Great to hear it!

**Comment by michaelckelly** [comment]
@guillelopez:

The fix for this issue was included in the `1.5.1` release. Thanks for your patience!
